### DISCUSSION TITLE
Configuration file to enable all protocols

### DISCUSSION THREAD

**Discussion by P-etr** [merbanan/rtl_433#2393] [Q&A] [answered]
Hi,

could you please upload me a configuration file that will enable all protocols? Use in Windows.
I wrote the rtl_433.conf file like this, but unfortunately it doesn't work:

```
-R

protocol 1
protocol 2
protocol 3
```

**Comment by zuckschwerdt** [comment, marked as answer]
All protocols are enabled by default. No need to give any `-R`. If you want additional protocols (*hint: you do not want that*) use e.g. `-R -1 -R 6 -R 7`

**Comment by P-etr** [comment]
Thank you for your quick help!
Why don't I want to enable all additional protocols?
I wanted to do a survey of all the transmitting devices in the area.

**Reply by zuckschwerdt** [reply]
All sensible protocols are enabled. The default-disabled protocols will not find more devices but only false matches.

**Reply by homeautomaton** [reply]
Hi Christian, I was just admiring your work on adding priorities so that more precise protocols can get a first shot at decoding before the sloppier ones run. I am experimenting with this and the conflict between Oregon V1 and Acurite Rain ( issue #353 ), which is what inspired the "disabled" feature, if I'm not mistaken.

I have both devices (and a ton of other 433Mhz devices), so plenty of false positives show up for me. Like the OP, I like to have all of the "good" protocols enabled so I can discover other devices transmitting in the area. As you know, turning on `-R 10` for my rain sensor disables everything else by default.

I've added in my own sandbox, an option `-k`, which "keeps" the other protocols when you add a specific `-R` option. I would gladly check that in and create a pull request if you think it would be accepted.

I have a question about improving the accuracy of that old Acurite Rain decoder, but I'll ask that in a different topic.

**Reply by zuckschwerdt** [reply]
re `-k`, use e.g. `-R -1 -R 10` to add a device.

**Reply by homeautomaton** [reply]
Excellent. I missed the nuance that your earlier advice removes a protocol first. I can't find that this behavior is clearly explained in the docs, but it's good to know. 

I'm using -R -10 -R 10, now, which also works, without the cost of removing some other protocol. It also moves the protocol to the end of the list, which, I guess, makes it later in the list of protocols to check!?! That's a nice side-effect, if so.

**Reply by homeautomaton** [reply]
I've submitted my first pull request: https://github.com/merbanan/rtl_433/pull/2430, which greatly improves the selectivity of the Acurite 896 Rain Gauge decoder. I've given it a priority of 10 and suggested we re-enable it, having tested with many other protocols in my environment, including the Oregon V1 sensors that are mis-categorized by the old code.
